### DISCUSSION TITLE
How to manage parts, which can be used for different applications? (3D Printer, Arduino projects and so on).

### DISCUSSION THREAD

**Discussion by miracuruzrh** [Part-DB/Part-DB-server#641] [Q&A]
Hi

I'm not 100% sure, how I should build up the structure of my database.
I have this use cases for my database:
Managing my inventory for Electronic parts and for mechanical parts.
The parts are used in DIY Arduino/Teensy projects and for my 3D Printers.

The difficulty starts there, when a part can be used for both of my use cases.
Let's say, I have a Microswitch, which can be used for 3D Printers and for DIY projects as well.

I thought that I can reflect this with the projects feature. But I'm not sure.
Because it is an optional part. I have currently "sensor less homing" where I don't need a microswitch.
But I can also change the setup to use a switch. It is like a optional part.

As far as I see, the project feature is static and doesn't allow to have something like variants or configurations.
Or would the solution for this to use the "Tag feature"?

I have many similar cases, where I'm not sure about this.
Another example is all the spare parts for printers I have. There I have different Hotends, which are replaceable together with other mounting parts and so on.

Maybe someone have experience, how to reflect such cases?

**Comment by virtadpt** [comment]
I use tags for this.
